Tests/Services/NfieldLocalUserServiceTests.cs

```diff
@@ -200,5 +200,68 @@ public async Task CanGetAllUsers()
             Assert.Equal("user2", user2.UserName);
             Assert.Null(user2.LastLogonDate);
         }
+
+        [Fact]
+        public async Task DownloadLogs()
+        {
+            const string activityId = "activity-id";
+            const string logsLink1 = "logs-link-1";
+            const string logsLink2 = "logs-link-2";
+            var query = new LogQueryModel
+            {
+                StartTime = DateTime.Now.Subtract(TimeSpan.FromDays(1)),
+                EndTime = DateTime.Now
+            };
+
+            _mockedHttpClient
+               .Setup(client => client.PostAsJsonAsync(new Uri(ServiceAddress, $"LocalUsersLogs"), It.Is<LogQueryModel>(
+                   q => q.StartTime == query.StartTime && q.EndTime == query.EndTime)))
+               .Returns(
+                Task.Factory.StartNew(
+                    () =>
+                    new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent(JsonConvert.SerializeObject(new { ActivityId = activityId }))
+                    })).Verifiable();
+
+            _mockedHttpClient
+                .Setup(client => client.GetAsync(new Uri(ServiceAddress, $"BackgroundActivities/{activityId}")))
+                .Returns(Task.Factory.StartNew(
+                    () =>
+                    new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent(JsonConvert.SerializeObject(new { DownloadDataUrl = logsLink1, ActivityId = activityId, Status = 2 /* Succeeded */ }))
+                    })).Verifiable();
+
+            // Test it using the model
+             var result = await _target.LogsAsync(query);
+            _mockedHttpClient.Verify();
+            Assert.Equal(logsLink1, result);
+
+            _mockedHttpClient
+                .Setup(client => client.PostAsJsonAsync(new Uri(ServiceAddress, $"LocalUsersLogs"), It.Is<LogQueryModel>(
+                    q => q.StartTime == query.StartTime && q.EndTime == query.EndTime)))
+                .Returns(
+                Task.Factory.StartNew(
+                    () =>
+                    new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent(JsonConvert.SerializeObject(new { ActivityId = activityId }))
+                    })).Verifiable();
```

Comment:
This setup is identical than the previous one.
You could verify that the calls to "LocalUserLogs" and "BackgroundActivities " separately instead use _mockedHttpClient.Verify();
I leave up to you.